flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/LocalExecutorITCase.java

```diff
@@ -151,6 +155,44 @@ public void testValidateSession() throws Exception {
 		assertEquals(expectedTables, actualTables);
 	}
 
+	@Test(timeout = 30_000L)
+	public void testDefaultCatalogDifferentFromBuiltin() throws Exception {
```

Comment:
Most IT cases here are testing runtime behavior, I'm not sure if we need to run a full IT case. But I'm fine if there is no better solution yet.

Reply:
I agree. I also don't like the test. I couldn't find a well isolated way to test this behavior though.